### PR TITLE
Normalize text in messages to reduce token length

### PR DESCRIPTION
**Pull request by Yun-BizQA**
## Purpose

* ...
* resolve the bug: [Rag sample clasic] Fix or revisit token estimation

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What to Check
Verify that the following are valid
* ...
* Korean and english words are running out of maximum token limit at the same time.

### CODE REVIEW

**Comment by pamelafox** [comment]
Thanks @Yun-BizQA ! Could you provide an example of Korean text that failed before? And then add a test for that? (See CONTRIBUTING.md for info on running the tests)

**Comment by Yun-BizQA** [comment]
The fix addresses the issue here: https://github.com/Azure-Samples/azure-search-openai-demo/issues/543
The maximum context length is reached easily for korean, chinese, japanese etc because python calculates their length wrong. The fix makes their len equal as the way English char counts, so conversation lasts longer before reaching to such err. The test is performed in the local deployment instead of codes.

File: app/backend/approaches/chatreadretrieveread.py [hunk 1]
```diff
@@ -259,21 +259,19 @@ def get_messages_from_history(
         for shot in few_shots:
             message_builder.append_message(shot.get("role"), shot.get("content"))
 
-        user_content = user_conv
         append_index = len(few_shots) + 1
 
         message_builder.append_message(self.USER, user_content, index=append_index)
 
         for h in reversed(history[:-1]):
+            if message_builder.token_length > max_tokens:
```
Comment (pamelafox) [inline app/backend/approaches/chatreadretrieveread.py:267]: Moved up to account for situation of running out of tokens with just the system prompt and user message

**Comment by pamelafox** [comment]
@Yun-BizQA Can you take a look at the changes I made and see if they make sense? I moved the normalization into the message builder so that its applied to all messages. I also added several tests. Let me know what you think.

also cc'ed @tonybaloney as he has experience with CJK language support.

**Comment by tonybaloney** [comment]
makes sense to me 👍🏻

**Review by tonybaloney — APPROVED** [review]

**Pull request merged**